Building master branch crashes on Apple M1

is building current Julia master crashing for anyone else? (i.e. commit aae68a5e030a04f3d4bfc31f4b30c002ddbc52ca, on : macOS 12.2 (21D49) on Apple M1 Max, Xcode 13.2.1).

It builds fine for me on Ubuntu x86.

```
    JULIA usr/lib/julia/sys-o.a
Generating REPL precompile statements... 36/36
Executing precompile statements... 1962/2006
Precompilation complete. Summary:
Total ───────  71.210485 seconds
Generation ──  52.778831 seconds 74.1167%
Execution ───  18.431653 seconds 25.8833%
LLVM ERROR: failed to perform tail call elimination on a call site marked musttail

signal (6): Abort trap: 6
in expression starting at none:0
__pthread_kill at /usr/lib/system/libsystem_kernel.dylib (unknown line)
Allocations: 158165722 (Pool: 158088219; Big: 77503); GC: 157
/bin/sh: line 1: 62460 Abort trap: 6           JULIA_BINDIR=/Users/mhorn/Projekte/Julia/julia.master/usr/bin WINEPATH="/Users/mhorn/Projekte/Julia/julia.master/usr/bin;$WINEPATH" /Users/mhorn/Projekte/Julia/julia.master/usr/bin/julia -O3 -C "apple-a12" --output-o /Users/mhorn/Projekte/Julia/julia.master/usr/lib/julia/sys-o.a.tmp --startup-file=no --warn-overwrite=yes --sysimage /Users/mhorn/Projekte/Julia/julia.master/usr/lib/julia/sys.ji /Users/mhorn/Projekte/Julia/julia.master/contrib/generate_precompile.jl 1
*** This error is usually fixed by running `make clean`. If the error persists, try `make cleanall`. ***
make[1]: *** [/Users/mhorn/Projekte/Julia/julia.master/usr/lib/julia/sys-o.a] Error 1
make: *** [julia-sysimg-release] Error 2
```

> [f090992](https://github.com/JuliaLang/julia/commit/f09099206c7d11ffbe25e3416d504dcbdd3de7ec) is the guilty commit according to bisect from #43852. Also a bunch of
> 
> ```
> ./julia.h:397:9: warning: anonymous types declared in an anonymous union are an extension [-Wnested-anon-types]
>        struct {
>        ^
> ```
> 
> warnings appeared.

Oh noooo, @Keno 😢 I actually tried to build the master branch just to test this PR 😆

I see a many more warnings with the latest commit on master (6c16f717f9871401eed9350f36cd84ab51778b72).

```
(base) gong@mac julia % make
    PERL base/pcre_h.jl
    PERL base/errno_h.jl
    PERL base/build_h.jl.phony
    PERL base/features_h.jl
    PERL base/features_h.jl
    PERL base/features_h.jl
    PERL base/file_constants.jl
    PERL base/uv_constants.jl
    PERL base/version_git.jl.phony
    CC cli/loader_exe.o
    CC cli/loader_lib.o
    CC cli/loader_trampolines.o
    LINK usr/lib/libjulia.1.8.dylib
    LINK usr/lib/libjulia.dylib
    LINK usr/bin/julia
    FLISP src/julia_flisp.boot
    FLISP src/julia_flisp.boot.inc
    CC src/jloptions.o
    CC src/runtime_ccall.o
In file included from /Users/gong/GitHub/julia/src/runtime_ccall.cpp:10:
./julia.h:397:9: warning: anonymous types declared in an anonymous union are an extension [-Wnested-anon-types]
        struct {
        ^
./julia.h:406:9: warning: anonymous types declared in an anonymous union are an extension [-Wnested-anon-types]
        struct {
        ^
2 warnings generated.
    CC src/rtutils.o
    CC src/jltypes.o
    CC src/gf.o
    CC src/typemap.o
    CC src/smallintset.o
    CC src/ast.o
    CC src/builtins.o
    CC src/module.o
    CC src/interpreter.o
    CC src/symbol.o
    CC src/dlload.o
    CC src/sys.o
    CC src/init.o
    CC src/task.o
    CC src/array.o
    CC src/dump.o
    CC src/staticdata.o
/Users/gong/GitHub/julia/src/staticdata.c:1709:33: warning: incompatible function pointer types passing 'void (jl_methtable_t *, void *)' (aka 'void (struct _jl_methtable_t *, void *)') to parameter of type 'int (*)(jl_methtable_t *, void *)' (aka 'int (*)(struct _jl_methtable_t *, void *)') [-Wincompatible-function-pointer-types]
    jl_foreach_reachable_mtable(set_nroots_sysimg_, NULL);
                                ^~~~~~~~~~~~~~~~~~
./julia_internal.h:615:39: note: passing argument to parameter 'visit' here
int jl_foreach_reachable_mtable(int (*visit)(jl_methtable_t *mt, void *env), void *env);
                                      ^
1 warning generated.
    CC src/toplevel.o
    CC src/jl_uv.o
    CC src/datatype.o
    CC src/simplevector.o
    CC src/runtime_intrinsics.o
    CC src/precompile.o
    CC src/threading.o
    CC src/partr.o
    CC src/stackwalk.o
    CC src/gc.o
    CC src/gc-debug.o
    CC src/gc-pages.o
    CC src/gc-stacks.o
    CC src/gc-alloc-profiler.o
In file included from /Users/gong/GitHub/julia/src/gc-alloc-profiler.cpp:3:
In file included from ./gc-alloc-profiler.h:6:
./julia.h:397:9: warning: anonymous types declared in an anonymous union are an extension [-Wnested-anon-types]
        struct {
        ^
./julia.h:406:9: warning: anonymous types declared in an anonymous union are an extension [-Wnested-anon-types]
        struct {
        ^
2 warnings generated.
    CC src/method.o
    CC src/jlapi.o
    CC src/signal-handling.o
/Users/gong/GitHub/julia/src/signal-handling.c:117:15: warning: unused function 'jl_safe_read_mem' [-Wunused-function]
static size_t jl_safe_read_mem(const volatile char *ptr, char *out, size_t len)
              ^
1 warning generated.
    CC src/safepoint.o
    CC src/timing.o
    CC src/subtype.o
    CC src/crc32c.o
'++crc' is not a recognized feature for this target (ignoring feature)
'++crc' is not a recognized feature for this target (ignoring feature)
    CC src/APInt-C.o
In file included from /Users/gong/GitHub/julia/src/APInt-C.cpp:10:
./julia.h:397:9: warning: anonymous types declared in an anonymous union are an extension [-Wnested-anon-types]
        struct {
        ^
./julia.h:406:9: warning: anonymous types declared in an anonymous union are an extension [-Wnested-anon-types]
        struct {
        ^
2 warnings generated.
    CC src/processor.o
In file included from /Users/gong/GitHub/julia/src/processor.cpp:10:
In file included from ./processor.h:5:
./julia.h:397:9: warning: anonymous types declared in an anonymous union are an extension [-Wnested-anon-types]
        struct {
        ^
./julia.h:406:9: warning: anonymous types declared in an anonymous union are an extension [-Wnested-anon-types]
        struct {
        ^
In file included from /Users/gong/GitHub/julia/src/processor.cpp:847:
./processor_arm.cpp:1815:40: warning: value size does not match register size specified by the constraint and modifier [-Wasm-operand-widths]
    asm volatile("mrs %0, fpcr" : "=r"(fpcr));
                                       ^
./processor_arm.cpp:1815:23: note: use constraint modifier "w"
    asm volatile("mrs %0, fpcr" : "=r"(fpcr));
                      ^~
                      %w0
./processor_arm.cpp:1821:40: warning: value size does not match register size specified by the constraint and modifier [-Wasm-operand-widths]
    asm volatile("msr fpcr, %0" :: "r"(fpcr));
                                       ^
./processor_arm.cpp:1821:29: note: use constraint modifier "w"
    asm volatile("msr fpcr, %0" :: "r"(fpcr));
                            ^~
                            %w0
4 warnings generated.
    CC src/ircode.o
    CC src/opaque_closure.o
    CC src/codegen-stubs.o
    CC src/coverage.o
In file included from /Users/gong/GitHub/julia/src/coverage.cpp:13:
./julia.h:397:9: warning: anonymous types declared in an anonymous union are an extension [-Wnested-anon-types]
        struct {
        ^
./julia.h:406:9: warning: anonymous types declared in an anonymous union are an extension [-Wnested-anon-types]
        struct {
        ^
2 warnings generated.
    LINK usr/lib/libjulia-internal.1.8.dylib
    LINK usr/lib/libjulia-internal.1.dylib
    LINK usr/lib/libjulia-internal.dylib
    CC src/codegen.o
In file included from /Users/gong/GitHub/julia/src/codegen.cpp:169:
./julia.h:397:9: warning: anonymous types declared in an anonymous union are an extension [-Wnested-anon-types]
        struct {
        ^
./julia.h:406:9: warning: anonymous types declared in an anonymous union are an extension [-Wnested-anon-types]
        struct {
        ^
2 warnings generated.
    CC src/llvm-ptls.o
In file included from /Users/gong/GitHub/julia/src/llvm-ptls.cpp:26:
./julia.h:397:9: warning: anonymous types declared in an anonymous union are an extension [-Wnested-anon-types]
        struct {
        ^
./julia.h:406:9: warning: anonymous types declared in an anonymous union are an extension [-Wnested-anon-types]
        struct {
        ^
2 warnings generated.
    CC src/jitlayers.o
In file included from /Users/gong/GitHub/julia/src/jitlayers.cpp:30:
./julia.h:397:9: warning: anonymous types declared in an anonymous union are an extension [-Wnested-anon-types]
        struct {
        ^
./julia.h:406:9: warning: anonymous types declared in an anonymous union are an extension [-Wnested-anon-types]
        struct {
        ^
/Users/gong/GitHub/julia/src/jitlayers.cpp:1016:4: warning: #warning is a language extension [-Wpedantic]
#  warning "JIT debugging (GDB integration) not available on LLVM < 14.0 (for JITLink)"
   ^
/Users/gong/GitHub/julia/src/jitlayers.cpp:1016:4: warning: "JIT debugging (GDB integration) not available on LLVM < 14.0 (for JITLink)" [-W#warnings]
4 warnings generated.
    CC src/aotcompile.o
In file included from /Users/gong/GitHub/julia/src/aotcompile.cpp:60:
./julia.h:397:9: warning: anonymous types declared in an anonymous union are an extension [-Wnested-anon-types]
        struct {
        ^
./julia.h:406:9: warning: anonymous types declared in an anonymous union are an extension [-Wnested-anon-types]
        struct {
        ^
2 warnings generated.
    CC src/debuginfo.o
In file included from /Users/gong/GitHub/julia/src/debuginfo.cpp:25:
./julia.h:397:9: warning: anonymous types declared in an anonymous union are an extension [-Wnested-anon-types]
        struct {
        ^
./julia.h:406:9: warning: anonymous types declared in an anonymous union are an extension [-Wnested-anon-types]
        struct {
        ^
2 warnings generated.
    CC src/disasm.o
In file included from /Users/gong/GitHub/julia/src/disasm.cpp:111:
./julia.h:397:9: warning: anonymous types declared in an anonymous union are an extension [-Wnested-anon-types]
        struct {
        ^
./julia.h:406:9: warning: anonymous types declared in an anonymous union are an extension [-Wnested-anon-types]
        struct {
        ^
2 warnings generated.
    CC src/llvm-simdloop.o
    CC src/llvm-muladd.o
In file included from /Users/gong/GitHub/julia/src/llvm-muladd.cpp:22:
./julia.h:397:9: warning: anonymous types declared in an anonymous union are an extension [-Wnested-anon-types]
        struct {
        ^
./julia.h:406:9: warning: anonymous types declared in an anonymous union are an extension [-Wnested-anon-types]
        struct {
        ^
2 warnings generated.
    CC src/llvm-final-gc-lowering.o
In file included from /Users/gong/GitHub/julia/src/llvm-final-gc-lowering.cpp:14:
In file included from ./codegen_shared.h:9:
./julia.h:397:9: warning: anonymous types declared in an anonymous union are an extension [-Wnested-anon-types]
        struct {
        ^
./julia.h:406:9: warning: anonymous types declared in an anonymous union are an extension [-Wnested-anon-types]
        struct {
        ^
2 warnings generated.
    CC src/llvm-pass-helpers.o
In file included from /Users/gong/GitHub/julia/src/llvm-pass-helpers.cpp:15:
In file included from ./codegen_shared.h:9:
./julia.h:397:9: warning: anonymous types declared in an anonymous union are an extension [-Wnested-anon-types]
        struct {
        ^
./julia.h:406:9: warning: anonymous types declared in an anonymous union are an extension [-Wnested-anon-types]
        struct {
        ^
2 warnings generated.
    CC src/llvm-late-gc-lowering.o
In file included from /Users/gong/GitHub/julia/src/llvm-late-gc-lowering.cpp:31:
In file included from ./codegen_shared.h:9:
./julia.h:397:9: warning: anonymous types declared in an anonymous union are an extension [-Wnested-anon-types]
        struct {
        ^
./julia.h:406:9: warning: anonymous types declared in an anonymous union are an extension [-Wnested-anon-types]
        struct {
        ^
2 warnings generated.
    CC src/llvm-lower-handlers.o
In file included from /Users/gong/GitHub/julia/src/llvm-lower-handlers.cpp:22:
./julia.h:397:9: warning: anonymous types declared in an anonymous union are an extension [-Wnested-anon-types]
        struct {
        ^
./julia.h:406:9: warning: anonymous types declared in an anonymous union are an extension [-Wnested-anon-types]
        struct {
        ^
2 warnings generated.
    CC src/llvm-gc-invariant-verifier.o
In file included from /Users/gong/GitHub/julia/src/llvm-gc-invariant-verifier.cpp:28:
In file included from ./codegen_shared.h:9:
./julia.h:397:9: warning: anonymous types declared in an anonymous union are an extension [-Wnested-anon-types]
        struct {
        ^
./julia.h:406:9: warning: anonymous types declared in an anonymous union are an extension [-Wnested-anon-types]
        struct {
        ^
2 warnings generated.
    CC src/llvm-propagate-addrspaces.o
In file included from /Users/gong/GitHub/julia/src/llvm-propagate-addrspaces.cpp:25:
In file included from ./codegen_shared.h:9:
./julia.h:397:9: warning: anonymous types declared in an anonymous union are an extension [-Wnested-anon-types]
        struct {
        ^
./julia.h:406:9: warning: anonymous types declared in an anonymous union are an extension [-Wnested-anon-types]
        struct {
        ^
2 warnings generated.
    CC src/llvm-multiversioning.o
In file included from /Users/gong/GitHub/julia/src/llvm-multiversioning.cpp:28:
./julia.h:397:9: warning: anonymous types declared in an anonymous union are an extension [-Wnested-anon-types]
        struct {
        ^
./julia.h:406:9: warning: anonymous types declared in an anonymous union are an extension [-Wnested-anon-types]
        struct {
        ^
2 warnings generated.
    CC src/llvm-alloc-opt.o
In file included from /Users/gong/GitHub/julia/src/llvm-alloc-opt.cpp:29:
In file included from ./codegen_shared.h:9:
./julia.h:397:9: warning: anonymous types declared in an anonymous union are an extension [-Wnested-anon-types]
        struct {
        ^
./julia.h:406:9: warning: anonymous types declared in an anonymous union are an extension [-Wnested-anon-types]
        struct {
        ^
2 warnings generated.
    CC src/llvm-alloc-helpers.o
In file included from /Users/gong/GitHub/julia/src/llvm-alloc-helpers.cpp:5:
In file included from ./codegen_shared.h:9:
./julia.h:397:9: warning: anonymous types declared in an anonymous union are an extension [-Wnested-anon-types]
        struct {
        ^
./julia.h:406:9: warning: anonymous types declared in an anonymous union are an extension [-Wnested-anon-types]
        struct {
        ^
2 warnings generated.
    CC src/cgmemmgr.o
In file included from /Users/gong/GitHub/julia/src/cgmemmgr.cpp:7:
./julia.h:397:9: warning: anonymous types declared in an anonymous union are an extension [-Wnested-anon-types]
        struct {
        ^
./julia.h:406:9: warning: anonymous types declared in an anonymous union are an extension [-Wnested-anon-types]
        struct {
        ^
2 warnings generated.
    CC src/llvm-remove-addrspaces.o
In file included from /Users/gong/GitHub/julia/src/llvm-remove-addrspaces.cpp:15:
In file included from ./codegen_shared.h:9:
./julia.h:397:9: warning: anonymous types declared in an anonymous union are an extension [-Wnested-anon-types]
        struct {
        ^
./julia.h:406:9: warning: anonymous types declared in an anonymous union are an extension [-Wnested-anon-types]
        struct {
        ^
2 warnings generated.
    CC src/llvm-remove-ni.o
In file included from /Users/gong/GitHub/julia/src/llvm-remove-ni.cpp:10:
./julia.h:397:9: warning: anonymous types declared in an anonymous union are an extension [-Wnested-anon-types]
        struct {
        ^
./julia.h:406:9: warning: anonymous types declared in an anonymous union are an extension [-Wnested-anon-types]
        struct {
        ^
2 warnings generated.
    CC src/llvm-julia-licm.o
In file included from /Users/gong/GitHub/julia/src/llvm-julia-licm.cpp:14:
./julia.h:397:9: warning: anonymous types declared in an anonymous union are an extension [-Wnested-anon-types]
        struct {
        ^
./julia.h:406:9: warning: anonymous types declared in an anonymous union are an extension [-Wnested-anon-types]
        struct {
        ^
2 warnings generated.
    CC src/llvm-demote-float16.o
    CC src/llvm-cpufeatures.o
In file included from /Users/gong/GitHub/julia/src/llvm-cpufeatures.cpp:26:
./julia.h:397:9: warning: anonymous types declared in an anonymous union are an extension [-Wnested-anon-types]
        struct {
        ^
./julia.h:406:9: warning: anonymous types declared in an anonymous union are an extension [-Wnested-anon-types]
        struct {
        ^
2 warnings generated.
    LINK usr/lib/libjulia-codegen.1.8.dylib
    LINK usr/lib/libjulia-codegen.1.dylib
    LINK usr/lib/libjulia-codegen.dylib
    JULIA usr/lib/julia/corecompiler.ji
Core.Compiler ──── 28.319 seconds
    JULIA usr/lib/julia/sys.ji
coreio.jl
exports.jl
essentials.jl
ctypes.jl
gcutils.jl
generator.jl
reflection.jl
options.jl
promotion.jl
tuple.jl
expr.jl
pair.jl
traits.jl
range.jl
error.jl
bool.jl
number.jl
int.jl
operators.jl
pointer.jl
refvalue.jl
refpointer.jl
checked.jl
strings/lazy.jl
indices.jl
array.jl
abstractarray.jl
subarray.jl
views.jl
baseext.jl
ntuple.jl
abstractdict.jl
iddict.jl
idset.jl
iterators.jl
namedtuple.jl
./build_h.jl
./version_git.jl
hashing.jl
rounding.jl
div.jl
float.jl
twiceprecision.jl
complex.jl
rational.jl
multinverses.jl
abstractarraymath.jl
arraymath.jl
simdloop.jl
reduce.jl
reshapedarray.jl
reinterpretarray.jl
bitarray.jl
bitset.jl
multimedia.jl
some.jl
dict.jl
abstractset.jl
set.jl
char.jl
strings/basic.jl
strings/string.jl
strings/substring.jl
osutils.jl
c.jl
io.jl
iobuffer.jl
intfuncs.jl
strings/strings.jl
strings/search.jl
strings/unicode.jl
strings/util.jl
strings/io.jl
regex.jl
pcre.jl
./pcre_h.jl
parse.jl
shell.jl
show.jl
compiler/ssair/show.jl
arrayshow.jl
methodshow.jl
cartesian.jl
multidimensional.jl
broadcast.jl
missing.jl
version.jl
sysinfo.jl
libc.jl
./errno_h.jl
libdl.jl
logging.jl
linked_list.jl
condition.jl
threads.jl
threadingconstructs.jl
atomics.jl
locks-mt.jl
lock.jl
channels.jl
task.jl
threads_overloads.jl
weakkeydict.jl
env.jl
binaryplatforms.jl
cpuid.jl
./features_h.jl
libuv.jl
./uv_constants.jl
asyncevent.jl
iostream.jl
stream.jl
filesystem.jl
path.jl
stat.jl
file.jl
./file_constants.jl
cmd.jl
process.jl
ttyhascolor.jl
secretbuffer.jl
randomdevice.jl
floatfuncs.jl
math.jl
special/cbrt.jl
special/exp.jl
special/hyperbolic.jl
special/trig.jl
special/rem_pio2.jl
special/log.jl
reducedim.jl
accumulate.jl
permuteddimsarray.jl
ordering.jl
sort.jl
fastmath.jl
Enums.jl
gmp.jl
ryu/Ryu.jl
ryu/utils.jl
ryu/shortest.jl
ryu/fixed.jl
ryu/exp.jl
mpfr.jl
combinatorics.jl
irrationals.jl
mathconstants.jl
meta.jl
stacktraces.jl
experimental.jl
opaque_closure.jl
deepcopy.jl
download.jl
summarysize.jl
errorshow.jl
initdefs.jl
threadcall.jl
uuid.jl
pkgid.jl
toml_parser.jl
loading.jl
timing.jl
util.jl
asyncmap.jl
deprecated.jl
docs/basedocs.jl
client.jl
docs/Docs.jl
docs/bindings.jl
docs/utils.jl
Base  ───────────── 14.178758 seconds
ArgTools  ─────────  2.531217 seconds
Artifacts  ────────  0.063541 seconds
Base64  ───────────  0.061001 seconds
CRC32c  ───────────  0.005136 seconds
FileWatching  ─────  0.065909 seconds
Libdl  ────────────  0.002190 seconds
Logging  ──────────  0.022526 seconds
Mmap  ─────────────  0.050645 seconds
NetworkOptions  ───  0.068903 seconds
SHA  ──────────────  0.108535 seconds
Serialization  ────  0.158214 seconds
Sockets  ──────────  0.192383 seconds
Unicode  ──────────  0.044128 seconds
DelimitedFiles  ───  0.061432 seconds
LinearAlgebra  ────  5.266818 seconds
Markdown  ─────────  0.439116 seconds
Printf  ───────────  0.088090 seconds
Random  ───────────  0.733423 seconds
Tar  ──────────────  0.166660 seconds
Dates  ────────────  0.909628 seconds
Distributed  ──────  0.440892 seconds
Future  ───────────  0.004470 seconds
InteractiveUtils  ─  0.240830 seconds
LibGit2  ──────────  0.816253 seconds
Profile  ──────────  0.274617 seconds
SparseArrays  ─────  1.947869 seconds
UUIDs  ────────────  0.010143 seconds
REPL  ─────────────  2.122485 seconds
SharedArrays  ─────  0.291869 seconds
Statistics  ───────  0.113225 seconds
SuiteSparse  ──────  0.901692 seconds
TOML  ─────────────  0.042251 seconds
Test  ─────────────  0.191280 seconds
LibCURL  ──────────  0.239219 seconds
Downloads  ────────  0.194281 seconds
Pkg  ──────────────  2.692248 seconds
LazyArtifacts  ────  0.002312 seconds
Stdlibs total  ──── 21.570620 seconds
Sysimage built. Summary:
Total ───────  35.750053 seconds 
Base: ───────  14.178758 seconds 39.6608%
Stdlibs: ────  21.570620 seconds 60.3373%
    JULIA usr/lib/julia/sys-o.a
Generating REPL precompile statements... 36/36
Executing precompile statements... 1966/2010
Precompilation complete. Summary:
Total ───────  64.233143 seconds
Generation ──  47.976136 seconds 74.6906%
Execution ───  16.257007 seconds 25.3094%
LLVM ERROR: failed to perform tail call elimination on a call site marked musttail

signal (6): Abort trap: 6
in expression starting at none:0
__pthread_kill at /usr/lib/system/libsystem_kernel.dylib (unknown line)
Allocations: 158354666 (Pool: 158277106; Big: 77560); GC: 159
/bin/sh: line 1: 18314 Abort trap: 6           JULIA_BINDIR=/Users/gong/GitHub/julia/usr/bin WINEPATH="/Users/gong/GitHub/julia/usr/bin;$WINEPATH" /Users/gong/GitHub/julia/usr/bin/julia -O3 -C "apple-a12" --output-o /Users/gong/GitHub/julia/usr/lib/julia/sys-o.a.tmp --startup-file=no --warn-overwrite=yes --sysimage /Users/gong/GitHub/julia/usr/lib/julia/sys.ji /Users/gong/GitHub/julia/contrib/generate_precompile.jl 1
*** This error is usually fixed by running `make clean`. If the error persists, try `make cleanall`. ***
make[1]: *** [/Users/gong/GitHub/julia/usr/lib/julia/sys-o.a] Error 1
make: *** [julia-sysimg-release] Error 2
```

The error message has evolved somewhat from the original, here is the latest build error:

```
(base) gong@mac julia % make
fatal: bad revision '^upstream/master'
fatal: Not a valid object name upstream/master
    JULIA usr/lib/julia/corecompiler.ji
LLVM ERROR: failed to perform tail call elimination on a call site marked musttail

signal (6): Abort trap: 6
in expression starting at compiler/bootstrap.jl:10
__pthread_kill at /usr/lib/system/libsystem_kernel.dylib (unknown line)
Allocations: 836107 (Pool: 836053; Big: 54); GC: 1
/bin/sh: line 1: 89473 Abort trap: 6           /Users/gong/GitHub/julia/usr/bin/julia -C "apple-a12" --output-ji /Users/gong/GitHub/julia/usr/lib/julia/corecompiler.ji.tmp --startup-file=no --warn-overwrite=yes -g0 -O0 compiler/compiler.jl
make[1]: *** [/Users/gong/GitHub/julia/usr/lib/julia/corecompiler.ji] Error 134
make: *** [julia-sysimg-ji] Error 2
```

--- a/src/ccall.cpp
+++ b/src/ccall.cpp
@@ -234,9 +234,10 @@ static GlobalVariable *emit_plt_thunk(
         // musttail support is very bad on ARM, PPC, PPC64 (as of LLVM 3.9)
         // Known failures includes vararg (not needed here) and sret.
 
-#if (defined(_CPU_X86_) || defined(_CPU_X86_64_) || (defined(_CPU_AARCH64_) && !defined(__APPLE__)))
+#if (defined(_CPU_X86_) || defined(_CPU_X86_64_) || (defined(_CPU_AARCH64_) && !defined(_OS_DARWIN_)))
         // Ref https://bugs.llvm.org/show_bug.cgi?id=47058
         // LLVM, as of 10.0.1 emits wrong/worse code when musttail is set
+        // Apple silicon macs give an LLVM ERROR if musttail is set here #44107. Purity PR
         if (!attrs.hasAttrSomewhere(Attribute::ByVal))
             ret->setTailCallKind(CallInst::TCK_MustTail);
 #endif

--- a/src/ccall.cpp
+++ b/src/ccall.cpp
@@ -233,8 +233,8 @@ static GlobalVariable *emit_plt_thunk(
     else {
         // musttail support is very bad on ARM, PPC, PPC64 (as of LLVM 3.9)
         // Known failures includes vararg (not needed here) and sret.
-#if (defined(_CPU_X86_) || defined(_CPU_X86_64_) || \
-                        defined(_CPU_AARCH64_))
+
+#if (defined(_CPU_X86_) || defined(_CPU_X86_64_) || (defined(_CPU_AARCH64_) && !defined(__APPLE__)))
         // Ref https://bugs.llvm.org/show_bug.cgi?id=47058
         // LLVM, as of 10.0.1 emits wrong/worse code when musttail is set
         if (!attrs.hasAttrSomewhere(Attribute::ByVal))

--- a/src/ccall.cpp
+++ b/src/ccall.cpp
@@ -237,7 +237,7 @@ static GlobalVariable *emit_plt_thunk(
 #if (defined(_CPU_X86_) || defined(_CPU_X86_64_) || (defined(_CPU_AARCH64_) && !defined(_OS_DARWIN_)))
         // Ref https://bugs.llvm.org/show_bug.cgi?id=47058
         // LLVM, as of 10.0.1 emits wrong/worse code when musttail is set
-        // Apple silicon macs give an LLVM ERROR if musttail is set here #44107. Purity PR
+        // Apple silicon macs give an LLVM ERROR if musttail is set here #44107.
         if (!attrs.hasAttrSomewhere(Attribute::ByVal))
             ret->setTailCallKind(CallInst::TCK_MustTail);
 #endif

